crates/common_enums/src/enums.rs

```diff
@@ -37,3 +38,2097 @@ pub enum CountryCode {
+        {
+            CountryAlpha2::deserialize(deserializer).map(Country::from_alpha2)
+        }
+    }
+
+    pub mod alpha3_country_code {
+        use std::fmt;
+
+        use serde::de::Visitor;
+
+        use super::*;
+
+        pub fn serialize<S>(code: &Country, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::Serializer,
+        {
+            code.to_alpha3().serialize(serializer)
+        }
+
+        struct FieldVisitor;
+
+        impl<'de> Visitor<'de> for FieldVisitor {
+            type Value = CountryAlpha3;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+                formatter.write_str("CountryAlpha3 as a string")
+            }
+        }
+
+        pub fn deserialize<'a, D>(deserializer: D) -> Result<Country, D::Error>
+        where
+            D: serde::Deserializer<'a>,
+        {
+            CountryAlpha3::deserialize(deserializer).map(Country::from_alpha3)
+        }
+    }
+
+    pub mod numeric_country_code {
+        use std::fmt;
+
+        use serde::de::Visitor;
+
+        use super::*;
+
+        pub fn serialize<S>(code: &Country, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::Serializer,
+        {
+            code.to_numeric().serialize(serializer)
+        }
+
+        struct FieldVisitor;
+
+        impl<'de> Visitor<'de> for FieldVisitor {
+            type Value = u32;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+                formatter.write_str("Numeric country code passed as an u32")
+            }
+        }
+
+        pub fn deserialize<'a, D>(deserializer: D) -> Result<Country, D::Error>
+        where
+            D: serde::Deserializer<'a>,
+        {
+            u32::deserialize(deserializer).map(Country::from_numeric)
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize)]
+struct Address {
+    #[serde(with = "custom_serde::alpha2_country_code")]
+    country: Country,
+}
+#[derive(serde::Serialize)]
+struct Alpha2Request {
+    #[serde(with = "custom_serde::alpha2_country_code")]
+    pub country: Country,
+}
+
+#[derive(serde::Serialize)]
+struct Alpha3Request {
+    #[serde(with = "custom_serde::alpha3_country_code")]
+    pub country: Country,
+}
+
+#[derive(serde::Serialize)]
+struct NumericRequest {
+    #[serde(with = "custom_serde::numeric_country_code")]
+    pub country: Country,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct HyperswitchRequestAlpha2 {
+    #[serde(with = "custom_serde::alpha2_country_code")]
+    pub country: Country,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct HyperswitchRequestAlpha3 {
+    #[serde(with = "custom_serde::alpha3_country_code")]
+    pub country: Country,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct HyperswitchRequestNumeric {
+    #[serde(with = "custom_serde::numeric_country_code")]
+    pub country: Country,
+}
```

Comment:
All these types could go in the `tests` module? I don't see them being used anywhere else in your PR.